### PR TITLE
Discrete CartPole Emitter-Receiver - fix coordinate system

### LINKED ISSUE #58: Several worlds still have old rotations
### CartPole Discrete
---
- [x] `cartPoleWorldEmitterReceiver.wbt` new fixed one works ok, but uses NUE coordinate system, instead of [ENU which is default](https://cyberbotics.com/doc/reference/worldinfo) (fix in #60)
- [x] `cartPoleWorldRobotSupervisor.wbt` is not updated at all - broken (fix in #66)
- [x] `cartPoleWorldRobotSupervisorStableBaselines.wbt` is not updated at all - broken (fix in #71)
---
### CartPole Continuous
- [x] `cartPoleWorldEmitterReceiver.wbt` is not updated at all - broken (fixed in #59)
- [x] `cartPoleWorldRobotSupervisor.wbt` is not updated at all - broken (fixed in #67)
---
### Find and Avoid
- [x] `smallWorld.wbt` is not updated at all, automatic rotation resolver partially fixes it (fix in #63 - Thank you @KelvinYang0320)
---
### KHR-3HV
- [x] `khr-3hv.wbt` is not updated at all, seems to be working correctly (fix in #76 , along with a wealth of quality improvements)
---
### Panda Goal Reaching
- [x] `pandaRL_training.wbt` is not updated at all, seems to be working correctly (fix in #61 - Thank you @KelvinYang0320)
---
### Pit Escape
- [x] `pit_escape.wbt` fixed in #56 and works ok, but still uses NUE coordinate system, instead of ENU (same as cartpole emitter-receiver discrete)

### PR DESCRIPTION
**Pull request by tsampazk**
Partial fix for issue #58

### CODE REVIEW

File: examples/cartpole/cartpole_discrete/controllers/supervisor_manager/supervisor_controller.py [hunk 1]
```diff
@@ -128,7 +128,7 @@ def is_done(self):
         if abs(poleAngle) > 0.261799388:  # 15 degrees off vertical
             return True
 
-        cartPosition = round(self.robot.getPosition()[2], 2)  # Position on z axis
+        cartPosition = round(self.robot.getPosition()[0], 2)  # Position on x axis
```
Comment (KelvinYang0320) [inline examples/cartpole/cartpole_discrete/controllers/supervisor_manager/supervisor_controller.py:131]: Change line 75~78 for `get_observations()`as well?
https://github.com/aidudezzz/deepworlds/blob/e9baef2bb658291020aed4ec28ad3a11f3ddfbd6/examples/cartpole/cartpole_discrete/controllers/supervisor_manager/supervisor_controller.py#L75-L78

**Review by KelvinYang0320 — CHANGES_REQUESTED** [review]
Thanks for fixing that.
I have left some comments.

File: examples/cartpole/cartpole_discrete/controllers/supervisor_manager/supervisor_controller.py (hunk 1 above)
Comment (tsampazk) [reply examples/cartpole/cartpole_discrete/controllers/supervisor_manager/supervisor_controller.py:131]: Oops, nice catch! I will update it soon, thanks :)

**Comment by KelvinYang0320** [comment]
@tsampazk 
It seems that with correct observation of cartPosition and cartVelocity, it takes more epochs to train the agent.
I change `episodeLimit` from 200 to 10000, but it still didn't reach a score of 195.0.
The results look fine, though. :thinking:

**Comment by tsampazk** [comment]
> @tsampazk It seems that with correct observation of cartPosition and cartVelocity, it takes more epochs to train the agent. I change `episodeLimit` from 200 to 10000, but it still didn't reach a score of 195.0. The results look fine, though.

Thanks for taking the time to test it! Out of the times i did a complete training run, in a lot of cases it can't reach the solved condition (which is quite harsh to be honest) in 2000 episodes,  but it seems to balance the pole nicely for each whole testing episode afterwards when it is deployed. Actually, the continuous case using DDPG is really slow to converge if it converges at all.

Regardless, we can stick to fixing whatever needs fixing in regards to the coordinate system here and we can look at the convergence issues at another time/issue/PR :P Feel free to open a new relevant issue to document your findings if you feel it is needed.

**Comment by KelvinYang0320** [comment]
> Regardless, we can stick to fixing whatever needs fixing in regards to the coordinate system here and we can look at the convergence issues at another time/issue/PR :P Feel free to open a new relevant issue to document your findings if you feel it is needed.

I open a new issue for that, so we can stick to the coordinate system here. Thanks!

File: examples/cartpole/cartpole_discrete/controllers/supervisor_manager/supervisor_controller.py (hunk 1 above)
Comment (KelvinYang0320) [reply examples/cartpole/cartpole_discrete/controllers/supervisor_manager/supervisor_controller.py:131]: Update these comments as well. Thanks!
https://github.com/aidudezzz/deepworlds/blob/e9baef2bb658291020aed4ec28ad3a11f3ddfbd6/examples/cartpole/cartpole_discrete/controllers/supervisor_manager/supervisor_controller.py#L22

https://github.com/aidudezzz/deepworlds/blob/e9baef2bb658291020aed4ec28ad3a11f3ddfbd6/examples/cartpole/cartpole_discrete/controllers/supervisor_manager/supervisor_controller.py#L42

**Comment by tsampazk** [comment]
@all-contributors please add @KelvinYang0320 for review

**Review by KelvinYang0320 — APPROVED** [review]
LGTM!

**Comment by KelvinYang0320** [comment]
@tsampazk    Could you choose "squash and merge" instead of "create a merge commit"?
There are more than one commit on the same file.

**Comment by KelvinYang0320** [comment]
> @all-contributors please add @KelvinYang0320 for review

@tsampazk Thank you!

**Comment by tsampazk** [comment]
> @tsampazk Could you choose "squash and merge" instead of "create a merge commit"? There are more than one commit on the same file.

We are using rebase by default which produces a much cleaner commit history without any merge commits :)

**Pull request merged**